cmd/cache-agent-refresh/main.go

```diff
@@ -0,0 +1,330 @@
+func checkNodeStatus(node string, port int) (bool, string) {
+	// Parse the response into NodeStatus struct
+	status, err := get_status(node, port)
+	if err != nil {
+		fmt.Printf("Error getting status from node %s:%d\n", node, port)
+		return false, ""
+	}
+
+	// Check if node is synced using the is_sync function with a tolerance of 2 layers
+	if is_sync(status.LatestLayer, status.ProcessedLayer, 2) {
+		fmt.Printf("Node %s:%d is online and synced\n", node, port)
+		return true, node
+	}
+	return false, ""
+}
+
+func refresh_cache(node string, port int, path string, interval int, prometheus string) error {
+	log.Printf("Refreshing cache for %s:%d%s\n", node, port, path)
+
+	last_refresh := prometheus_metrics_parcer(prometheus, "cache_agent_last_refresh", path)
+	if last_refresh < float64(interval*60) {
+		return nil
+	}
+
+	url := fmt.Sprintf("%s:%d%s", node, port, path)
+
+	// HTTP GET request to refresh cache
+	resp, err := http.Get(url)
+	if err != nil || resp.StatusCode != 200 {
+		fmt.Printf("Error refreshing cache for %s:%d%s, status code: %d\n", node, port, path, resp.StatusCode)
+		return err
+	}
+	defer resp.Body.Close()
+
+	log.Printf("Cache refreshed for %s:%d%s\n", node, port, path)
+	return nil
+}
+
+func epoch_replace(path string, epoch int) string {
+	// Replace the :id and :epoch with the current epoch
+	epoch_path := strings.ReplaceAll(path, ":id", strconv.Itoa(epoch))
+	epoch_path = strings.ReplaceAll(epoch_path, ":epoch", strconv.Itoa(epoch))
+	return epoch_path
+}
+
+var flags = []cli.Flag{
+	&cli.IntFlag{
+		Name:        "layers-per-epoch",
+		Usage:       "Number of layers per epoch",
+		Required:    false,
+		Destination: &layersPerEpoch,
+		Value:       4032,
+		EnvVars:     []string{"SPACEMESH_LAYERS_PER_EPOCH"},
+	},
+	&cli.StringFlag{
+		Name:        "target-nodes",
+		Usage:       "Comma separated list of target nodes",
+		Required:    true,
+		Destination: &targetNodesFlag,
+		EnvVars:     []string{"TARGET_NODES"},
+	},
+	&cli.IntFlag{
+		Name:        "target-nodes-json-port",
+		Usage:       "Port for the JSON API of the target nodes",
+		Required:    true,
+		Destination: &targetNodesJsonPortFlag,
+		EnvVars:     []string{"TARGET_NODES_JSON_PORT"},
+	},
+	&cli.IntFlag{
+		Name:        "target-nodes-refresh-port",
+		Usage:       "Port for the refresh API of the target nodes",
+		Required:    true,
+		Destination: &targetNodesRefreshPortFlag,
+		EnvVars:     []string{"TARGET_NODES_REFRESH_PORT"},
+	},
+	&cli.StringFlag{
+		Name:        "target-nodes-refresh-metric",
+		Usage:       "Port for the refresh API of the target nodes",
+		Required:    true,
+		Destination: &targetNodesRefreshMetricFlag,
+		EnvVars:     []string{"TARGET_NODES_REFRESH_METRIC"},
+	},
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "cache-agent-refresh"
+	app.Version = fmt.Sprintf("%s, commit '%s', branch '%s'", version, commit, branch)
+	app.Flags = flags
+	app.Writer = os.Stderr
+
+	app.Action = func(ctx *cli.Context) error {
```

Comment:
It would improve the readability if you would extract this whole block as a normal named function.